### PR TITLE
Make uid/gid configurable & change group of files - part 3

### PR DESCRIPTION
**Pull request by lfrancke**
# Description

This is a follow-up for #849 and #890 and includes:

Integration tests ran locally successfully when the :heavy_check_mark: is added

- statsd_exporter
- Superset :heavy_check_mark: 
- Trino :heavy_check_mark: (did https://github.com/stackabletech/trino-operator/pull/652 as part of this)
- Trino CLI
- Vector
- ZooKeeper :heavy_check_mark:  (did https://github.com/stackabletech/zookeeper-operator/pull/868 as part of this)

### CODE REVIEW

File: superset/Dockerfile [hunk 1]
```diff
@@ -105,22 +106,37 @@ ENV FLASK_APP="superset.app:create_app()" \
...
-    rm -rf /var/cache/yum
+RUN <<EOF
+microdnf update
+microdnf install \
+  cyrus-sasl \
+  openldap \
+  openldap-clients \
+  openssl-libs \
+  openssl-pkcs11 \
+  python${PYTHON}
```
Comment (NickLarsenNZ) [inline superset/Dockerfile:117]:
```suggestion
  "python${PYTHON}"
```

File: superset/Dockerfile [hunk 2]
```diff
@@ -105,22 +106,37 @@ ENV FLASK_APP="superset.app:create_app()" \
...
+
+microdnf clean all
+rm -rf /var/cache/yum
+
+# All files and folders owned by root group to support running as arbitrary users.
+# This is best practice as all container users will belong to the root group (0).
+chown -R ${STACKABLE_USER_UID}:0 /stackable
```
Comment (NickLarsenNZ) [inline superset/Dockerfile:124]:
```suggestion
chown -R "${STACKABLE_USER_UID}:0" /stackable
```

Comment (NickLarsenNZ) [reply superset/Dockerfile:124]: Same applies to any others.
Feel free to ignore if the previous changes also didn't do this. We can clean it up in a later PR.

File: tools/Dockerfile [hunk 3]
```diff
@@ -30,16 +31,30 @@ RUN microdnf update && \
...
-RUN curl https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/${TARGETARCH}/kubectl \
-    -o /stackable/bin/kubectl && chmod +x /stackable/bin/kubectl
-
-RUN curl https://github.com/stedolan/jq/releases/download/jq-${JQ_VERSION}/jq-linux64 \
-    -o /stackable/bin/jq && \
-    chmod +x /stackable/bin/jq
-
-USER stackable
+RUN <<EOF
+curl https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/${TARGETARCH}/kubectl \
```
Comment (NickLarsenNZ) [inline tools/Dockerfile:39]:
```suggestion
curl "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/${TARGETARCH}/kubectl" \
```

File: tools/Dockerfile [hunk 4]
```diff
@@ -30,16 +31,30 @@ RUN microdnf update && \
...
+    -o /stackable/bin/kubectl
+chmod +x /stackable/bin/kubectl
+
+curl https://github.com/stedolan/jq/releases/download/jq-${JQ_VERSION}/jq-linux64 \
```
Comment (NickLarsenNZ) [inline tools/Dockerfile:43]:
```suggestion
curl "https://github.com/stedolan/jq/releases/download/jq-${JQ_VERSION}/jq-linux64" \
```

File: trino-cli/Dockerfile [hunk 5]
```diff
@@ -22,14 +23,27 @@ RUN microdnf update && \
...
 
-COPY --chown=stackable:stackable trino-cli/licenses /licenses
+COPY --chown=${STACKABLE_USER_UID}:0 trino-cli/licenses /licenses
 
 WORKDIR /stackable/trino-cli
 
-RUN curl -O https://repo.stackable.tech/repository/packages/trino-cli/trino-cli-${PRODUCT}-executable.jar \
-    && ln -s trino-cli-${PRODUCT}-executable.jar trino-cli-executable.jar
+RUN <<EOF
+curl -O https://repo.stackable.tech/repository/packages/trino-cli/trino-cli-${PRODUCT}-executable.jar
```
Comment (NickLarsenNZ) [inline trino-cli/Dockerfile:32]:
```suggestion
curl -O "https://repo.stackable.tech/repository/packages/trino-cli/trino-cli-${PRODUCT}-executable.jar"
```

File: trino-cli/Dockerfile [hunk 6]
```diff
@@ -22,14 +23,27 @@ RUN microdnf update && \
...
+ln -s trino-cli-${PRODUCT}-executable.jar trino-cli-executable.jar
```
Comment (NickLarsenNZ) [inline trino-cli/Dockerfile:33]:
```suggestion
ln -s "trino-cli-${PRODUCT}-executable.jar" trino-cli-executable.jar
```

**Review by NickLarsenNZ — APPROVED** [review]
LGTM.

Left comments about things that should be picked up by hadolint (shellcheck), but then stopped. I can comment on the rest if you want, or we can leave those changes for another time.

File: superset/Dockerfile (hunk 2 above)
Comment (lfrancke) [reply superset/Dockerfile:124]: The previous changes also didn't do this but that doesn't mean we can't do it here. At least for your suggestions as that's already done.

Comment (lfrancke) [reply superset/Dockerfile]: That said: These are replaced at the "Docker" level before it even gets to bash so I don't think it's super important in this case.

Comment (NickLarsenNZ) [reply superset/Dockerfile]: Oh, good point!

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**